core/selection.cpp

```diff
@@ -231,18 +222,47 @@ bool setSelection(const std::vector<dive *> &selection, dive *currentDive, int c
 	if (currentDc >= 0)
 		dc_number = currentDc;
 	fixup_current_dc();
+	return divesToSelect;
+}
 
-	// Send the new selection
-	emit diveListNotifier.divesSelected(divesToSelect);
-	return current_dive != oldCurrent;
+static void clear_trip_selection()
+{
+	amount_trips_selected = 0;
+	for (int i = 0; i < trip_table.nr; ++i)
+		trip_table.trips[i]->selected = false;
 }
 
-bool setSelection(const std::vector<dive *> &selection)
+// Reset the selection to the dives of the "selection" vector and send the appropriate signals.
+// Set the current dive to "currentDive". "currentDive" must be an element of "selection" (or
+// null if "selection" is empty).
+void setSelection(const std::vector<dive *> &selection, dive *currentDive, int currentDc)
```

Comment:
it is ironic that it was just the previous commit that made this function `bool` - but I think you are doing this in a methodical fashion here, so I'm not arguing against it.

Reply:
Yes, the history is a mess. One could probably squeeze a few of these things.